docs/guides/Uniq Factory Management/factory-purchase-options-examples.md

```diff
@@ -0,0 +1,330 @@
+        "transfer_tokens_receiver_account": null,
+        "factories": [{
+            "token_factory_id": 42,
+            "count": 1,
+            "strategy": 0
+        }]
+    },
+    "sale_shares": [],
+    "maximum_uos_payment": null,
+    "group_restriction": null,
+    "purchase_window_start": null,
+    "purchase_window_end": null,
+    "memo": ""
+  }
+]' -p factory.manager
+```
+:::
+
+When purchasing the transaction needs to specify which token exactly the user shows as a proof of satisfying condition of ownership for the token from factory 42. In this case assume token 77 was minted from factory 42
+
+::: details purchase.a
+```sh
+cleos push action eosio.nft.ft purchase.a '[
+  {
+    "token_factory_id": 100,
+    "index": 0,
+    "max_price": "100.00000000 UOS",
+    "buyer": "alice",
+    "receiver": "alice",
+    "promoter_id": null,
+    "user_uniqs": {
+      "tokens": [{
+        "token_id": 77,
+        "strategy": 0
+      }]
+    },
+    "memo": ""
+  }
+]' -p alice
+```
+:::
+
+## Exclusive access to purchase option via user groups
+
+Alternative condition for allowing direct purchases from the factory can be the usage of user groups ([covered here](../../contracts/User%20Group%20Contract/overview.md)). In this case user must belong to certain group(s) or not be a part of a specific group(s).
+
+Example below covers the simplest case where user must belong to the user groups with IDs 11 and 12 at the same time. For more advanced usage reference the action documentation: [setprchsreq.a user groups support](../../contracts/NFT%20Contract/NFT%20Actions/setprchsreq.a.md#example-usage-of-the-parameter-group-restriction)
+
+::: details setprchsreq.a
+```sh
+cleos push action eosio.nft.ft setprchsreq.a '[
+  {
+    "token_factory_id": 100,
+    "index": 0,
+    "price": "50.00000000 UOS",
+    "purchase_limit": null,
+    "promoter_basis_point": 100,
+    "purchase_option_with_uniqs": null,
+    "sale_shares": [],
+    "maximum_uos_payment": null,
+    "group_restriction": [11, 12],
+    "purchase_window_start": null,
+    "purchase_window_end": null,
+    "memo": ""
+  }
+]' -p factory.manager
+```
+:::
+
+When purchasing no extra input is required from the user, the verification of groups membership will be verified by the smart contract automatically
+
+::: details purchase.a
+```sh
+cleos push action eosio.nft.ft purchase.a '[
+  {
+    "token_factory_id": 100,
+    "index": 0,
+    "max_price": "100.00000000 UOS",
+    "buyer": "alice",
+    "receiver": "alice",
+    "promoter_id": null,
+    "user_uniqs": null,
+    "memo": ""
+  }
+]' -p alice
+```
+:::
+
+## Using purchase option for swapping
+
+"Swapping" in this case implies the process where user looses ownership of his Uniq, the Uniq gets destroyed in the process and user gets a new Uniq from the factory instead. Example below requires user to give up two Uniqs: one from factory 43 and one from factory 44. Note how `strategy` is set to 1
```

Comment:
Can we also mention the uniq without any cost?
Like it is possible to do this swapping only with uniqs. no need to pay anything.